syft/pkg/cataloger/golang/exe.go

```diff
@@ -57,23 +60,35 @@ func openExe(file io.ReadCloser) (exe, error) {
 			return nil, err
 		}
 
-		return &elfExe{file, e}, nil
+		return []exe{&elfExe{file, e}}, nil
 	}
 
 	if bytes.HasPrefix(data, []byte("MZ")) {
 		e, err := pe.NewFile(f)
 		if err != nil {
 			return nil, err
 		}
-		return &peExe{file, e}, nil
+		return []exe{&peExe{file, e}}, nil
 	}
 
 	if bytes.HasPrefix(data, []byte("\xFE\xED\xFA")) || bytes.HasPrefix(data[1:], []byte("\xFA\xED\xFE")) {
 		e, err := macho.NewFile(f)
 		if err != nil {
 			return nil, err
 		}
-		return &machoExe{file, e}, nil
+		return []exe{&machoExe{file, e}}, nil
+	}
+
+	if bytes.HasPrefix(data, []byte("\xCA\xFE\xBA\xBE")) || bytes.HasPrefix(data[1:], []byte("\xCA\xFE\xBA\xBF")) {
```

Comment:
It'd be good to add a sense of meaning to this `if` line in order to motivate these `bytes.HasPrefix` calls for the reader. If these conditions "mean" something, that'd be great as a factored out function, named for what it's doing. WDYT?

Reply:
I can do that for the new code added, however, for code that was copied out of the stdlib I have a preference to leave as is (otherwise when we incorporate new stdlib code we need to re-re-factor it, which isn't fun).